maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/NodeHelper.java

```diff
@@ -32,19 +36,38 @@ public class NodeHelper {
 	
 	private final BuildPluginManager pluginManager;
 
-	private Path tempDir;
-	
-	public NodeHelper(MavenProject mavenProject,
-			MavenSession mavenSession,
-			BuildPluginManager pluginManager) {
+	private final Path tempDir;
+
+	private static final Logger logger = LoggerFactory.getLogger(NodeHelper.class);
+
+	public Path getTempDir() {
+		return tempDir;
+	}
+
+	public NodeHelper(MavenProject mavenProject, MavenSession mavenSession, BuildPluginManager pluginManager) {
 		this.mavenProject = mavenProject;
 		this.mavenSession = mavenSession;
 		this.pluginManager = pluginManager;
 		try {
-			this.tempDir = Files.createTempDirectory("openmrs-sdk-node");
-			this.tempDir.toFile().deleteOnExit();
+			Properties sdkProperties = getSdkProperties();
```

Comment:
It's preferrable, I think to move this code into `getTempDir()` that way we can defer creating a temporary directory until we actually need it.